### PR TITLE
PAS-384 | Fix delete passkey bug once and for all

### PR DESCRIPTION
**Pull request by jonashendrickx**
### Ticket

- Closes [PAS-384](https://bitwarden.atlassian.net/browse/PAS-384)

### Description

We were unable to delete passkeys on the profile page, because it was trying to use the `IScopedPasswordlessClient`. We're now injecting the `IPasswordlessClient` or its derivative `IScopedPasswordlessClient` using the `PasswordlessClient` parameter on the`Credentials.razor` component. A second parameter was then created to pass the `UserId`.

Similarly the user is passed from the claim, rather than using the expensive `UserManager<ConsoleAdmin>` calling the database with an additional trip. The credentials for a user in a given app for a customer are passed using a query parameter `userId` this behavior was not changed.

### Shape

See above.

### Screenshots

n/a

### Checklist
I did the following to ensure that my changes were tested thoroughly:
- [ ] Test all 3 screens where credentials were present

I did the following to ensure that my changes did not introduce new security vulnerabilities:
- [ ] Reduced security impact where user id can be obtained from the claim directly for a logged-in admin

[PAS-384]: https://bitwarden.atlassian.net/browse/PAS-384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/AdminConsole/Pages/Account/Profile.cshtml.cs [hunk 1]
```diff
@@ -1,39 +1,9 @@
...
-
-    public async Task OnGet()
-    {
-        var userId = _userManager.GetUserId(HttpContext.User);
-        if (userId != null)
-        {
-            Credentials = await _client.ListCredentialsAsync(userId);
-        }
-    }
+    public string UserId => HttpContext.User.Claims.Single(c => c.Type == ClaimTypes.NameIdentifier).Value;
```
Comment (abergs) [inline src/AdminConsole/Pages/Account/Profile.cshtml.cs:8]: suggest, separate issue: It would be nice to have an extension method for this.

Comment (jonashendrickx) [reply src/AdminConsole/Pages/Account/Profile.cshtml.cs]: @abergs we apparently had one already

File: src/AdminConsole/Components/Shared/Credentials.razor.cs [hunk 2]
```diff
@@ -46,11 +42,18 @@ public IReadOnlyCollection<CredentialModel> GetItems()
     [Parameter]
     public bool HideDetails { get; set; }
 
+    [Parameter]
+    public required IPasswordlessClient PasswordlessClient { get; set; }
```
Comment (abergs) [inline src/AdminConsole/Components/Shared/Credentials.razor.cs:46]: question: Why doesn't it work with the injected one, why do we need to pass this as a parameter?

**Review by abergs — APPROVED** [review]
shape, tech: LGTM

File: src/AdminConsole/Components/Shared/Credentials.razor.cs (hunk 2 above)
Comment (jonashendrickx) [reply src/AdminConsole/Components/Shared/Credentials.razor.cs:46]: This component is used on 3 pages in total:

- /account/useronboarding: uses `IPasswordlessClient`
- /account/profile: uses `IPasswordlessClient`
- /app/<appid>/credentials/list: uses `IScopedPasswordlessClient` (which replaces the ApiSecret from admin app with the one from the customer's app)

**Pull request merged**